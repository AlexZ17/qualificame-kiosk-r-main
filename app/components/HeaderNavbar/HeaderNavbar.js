import React, { Component } from 'react';
import { 
	Text, 
	View,
	SafeAreaView,
	StatusBar 
} from 'react-native';
import PropTypes from 'prop-types';
import { LinearGradient } from 'expo-linear-gradient';
import styles from './HeaderNavbarStyle';

export default class HeaderNavbar extends Component {
	render() {
		let gradientColors;
		switch(this.props.bgColor){
			case 'green':
				gradientColors = ['#0F9F92', '#12B184'];
			break;
			case 'blue':
				gradientColors = ['#154BBA', '#107EC2'];
			break;
			case 'orange':
				gradientColors = ['#E62808', '#FC680A'];
			break;
			case 'pink':
				gradientColors = ['#E31862', '#EB194D'];
			break;
			default:
				gradientColors = ['#fff', '#fff'];
		}

		let content = (
			<View style={[(!this.props.unSet && styles.navBarHeight), styles.above, styles.row, styles.alignItemsCenter, styles.statusBarSpace, styles.baseHorizontalMargin]}>
				<View style={[styles.flex1, styles.behind]}>
					{this.props.left && 
						this.props.left
					}
				</View>
				{this.props.center && 
					<View style={[styles.flex2, styles.alignItemsCenter]}>
						{this.props.center}
					</View>
				}
				<View style={[styles.flex1, styles.alignItemsFlexEnd ]}>
					{this.props.right && 
						this.props.right
					}
				</View>
			</View>
		)

		return (
			<SafeAreaView>
				<StatusBar barStyle={this.props.statusBar} />
				{ !!this.props.bgColor ?
					<LinearGradient colors={gradientColors}  start={[0, 0]} end={[1, 0]}>
						{content}
					</LinearGradient>
				:
					<React.Fragment>
						{content}
					</React.Fragment>				
				}
			</SafeAreaView>
		);
	}
}

	HeaderNavbar.propTypes = {
		statusBar: PropTypes.string,
		left: PropTypes.object,
		center: PropTypes.object,
		right: PropTypes.object,
		bgColor: PropTypes.string,
		unSet: PropTypes.bool,
	}

	HeaderNavbar.defaultProps = {
		statusBar: 'default',
		left: null,
		center: null,
		right: null,
		bgColor: "",
		unSet: false,
	}
