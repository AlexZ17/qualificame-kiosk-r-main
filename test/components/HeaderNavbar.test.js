import React from 'react';
import { shallow } from 'enzyme';
import {
	SafeAreaView,
	StatusBar 
} from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';

import HeaderNavbar from 'qualificame-kiosk/app/components/HeaderNavbar';

function createTestProps (props) {
	return {
		statusBar: 'default',
		left: null,
		center: null,
		right: null,
		bgColor: "",
		unSet: false,
		...props
	}
}

describe('rendering', () => {
	let wrapper;
	const createWrapper = props => shallow(<HeaderNavbar {...props}/>);
	beforeEach(() => {
		const props = createTestProps()
		wrapper = createWrapper(props);
	});
	it('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});
	it('displays SafeAreaView', () => {
		expect(wrapper.find(SafeAreaView).exists()).toBeTruthy();
	});
	it('displays StatusBar', () => {
		expect(wrapper.find(StatusBar).exists()).toBeTruthy();
	});
	it('displays feather icon', () => {
		expect(wrapper.find(LinearGradient).exists()).toBe(false);
	});
	it('renders StatusBar with correct barStyle', () => {
		expect(wrapper.find(StatusBar).prop('barStyle') ).toEqual('default');
	});
})