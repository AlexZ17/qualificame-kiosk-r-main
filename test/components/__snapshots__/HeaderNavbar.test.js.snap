// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders as expected 1`] = `
<SafeAreaView>
  <StatusBar
    animated={false}
    barStyle="default"
    showHideTransition="fade"
  />
  <Component
    style={
      Array [
        Object {
          "height": 64,
        },
        undefined,
        Object {
          "flexDirection": "row",
        },
        Object {
          "alignItems": "center",
        },
        Object {
          "paddingTop": 0,
        },
        Object {
          "marginHorizontal": 16,
        },
      ]
    }
  >
    <Component
      style={
        Array [
          Object {
            "flex": 1,
          },
          undefined,
        ]
      }
    />
    <Component
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "alignItems": "flex-end",
          },
        ]
      }
    />
  </Component>
</SafeAreaView>
`;
